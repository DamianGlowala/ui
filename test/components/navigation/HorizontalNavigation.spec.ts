import { describe, it, expect } from 'vitest'
import { UHorizontalNavigation } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'
import ui from '../../../src/runtime/ui.config/navigation/horizontalNavigation'

describe('HorizontalNavigation', () => {
  it.each([
    ['basic case', {}],
    ['with links', { props: { links: [ { label: 'Home', labelClass: 'text-sm', icon: 'i-heroicons-academic-cap', iconClass: 'w-5 h-5' }, { label: 'Settings', badge: 'badge' },{ label: 'About', avatar: { src: 'https://repository-images.githubusercontent.com/428329515/43fec891-9030-4601-8233-5d45ba5c6013', alt: 'nuxtui' }  }], ui: { base: ui.base, icon: { base: ui.icon.base }, avatar: { base: ui.avatar.base }, badge: { base: ui.badge.base } } } }],
    ['with avatar slot', { slots: { avatar: 'Avatar slot' }, props: { links: [ { label: 'Home', labelClass: 'text-sm', icon: 'i-heroicons-academic-cap', iconClass: 'w-5 h-5' }, { label: 'Settings', badge: 'badge' },{ label: 'About', avatar: { src: 'https://repository-images.githubusercontent.com/428329515/43fec891-9030-4601-8233-5d45ba5c6013', alt: 'nuxtui' }  }], ui: { base: ui.base, icon: { base: ui.icon.base }, avatar: { base: ui.avatar.base }, badge: { base: ui.badge.base } } } }],
    ['with icon slot', { slots: { avatar: 'Icon slot' }, props: { links: [ { label: 'Home', labelClass: 'text-sm', icon: 'i-heroicons-academic-cap', iconClass: 'w-5 h-5' }, { label: 'Settings', badge: 'badge' },{ label: 'About', avatar: { src: 'https://repository-images.githubusercontent.com/428329515/43fec891-9030-4601-8233-5d45ba5c6013', alt: 'nuxtui' }  }], ui: { base: ui.base, icon: { base: ui.icon.base }, avatar: { base: ui.avatar.base }, badge: { base: ui.badge.base } } } }],
    ['with badge slot', { slots: { avatar: 'Badge slot' }, props: { links: [ { label: 'Home', labelClass: 'text-sm', icon: 'i-heroicons-academic-cap', iconClass: 'w-5 h-5' }, { label: 'Settings', badge: 'badge' },{ label: 'About', avatar: { src: 'https://repository-images.githubusercontent.com/428329515/43fec891-9030-4601-8233-5d45ba5c6013', alt: 'nuxtui' }  }], ui: { base: ui.base, icon: { base: ui.icon.base }, avatar: { base: ui.avatar.base }, badge: { base: ui.badge.base } } } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof HorizontalNavigation.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UHorizontalNavigation)
    expect(html).toMatchSnapshot()
  })
})
