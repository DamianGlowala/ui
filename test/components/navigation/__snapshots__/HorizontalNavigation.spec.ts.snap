// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HorizontalNavigation > renders basic case correctly 1`] = `
"<nav>
  <ul class=""></ul>
</nav>"
`;

exports[`HorizontalNavigation > renders with avatar slot correctly 1`] = `
"<nav>
  <ul class="">
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75">Avatar slot<span class="i-heroicons-academic-cap flex-shrink-0 w-5 h-5"></span><span class="text-sm"><!--v-if--> Home</span>
        <!--v-if-->
      </button></li>
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75">Avatar slot
        <!--v-if--><span class=""><!--v-if--> Settings</span><span class="inline-flex items-center font-medium text-xs px-2 py-1 bg-primary-500 dark:bg-primary-400 text-white dark:text-gray-900 flex-shrink-0 ml-auto relative rounded">badge</span>
      </button></li>
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75">Avatar slot
        <!--v-if--><span class=""><!--v-if--> About</span>
        <!--v-if-->
      </button></li>
  </ul>
</nav>"
`;

exports[`HorizontalNavigation > renders with badge slot correctly 1`] = `
"<nav>
  <ul class="">
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75">Badge slot<span class="i-heroicons-academic-cap flex-shrink-0 w-5 h-5"></span><span class="text-sm"><!--v-if--> Home</span>
        <!--v-if-->
      </button></li>
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75">Badge slot
        <!--v-if--><span class=""><!--v-if--> Settings</span><span class="inline-flex items-center font-medium text-xs px-2 py-1 bg-primary-500 dark:bg-primary-400 text-white dark:text-gray-900 flex-shrink-0 ml-auto relative rounded">badge</span>
      </button></li>
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75">Badge slot
        <!--v-if--><span class=""><!--v-if--> About</span>
        <!--v-if-->
      </button></li>
  </ul>
</nav>"
`;

exports[`HorizontalNavigation > renders with icon slot correctly 1`] = `
"<nav>
  <ul class="">
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75">Icon slot<span class="i-heroicons-academic-cap flex-shrink-0 w-5 h-5"></span><span class="text-sm"><!--v-if--> Home</span>
        <!--v-if-->
      </button></li>
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75">Icon slot
        <!--v-if--><span class=""><!--v-if--> Settings</span><span class="inline-flex items-center font-medium text-xs px-2 py-1 bg-primary-500 dark:bg-primary-400 text-white dark:text-gray-900 flex-shrink-0 ml-auto relative rounded">badge</span>
      </button></li>
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75">Icon slot
        <!--v-if--><span class=""><!--v-if--> About</span>
        <!--v-if-->
      </button></li>
  </ul>
</nav>"
`;

exports[`HorizontalNavigation > renders with links correctly 1`] = `
"<nav>
  <ul class="">
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75">
        <!--v-if--><span class="i-heroicons-academic-cap flex-shrink-0 w-5 h-5"></span><span class="text-sm"><!--v-if--> Home</span>
        <!--v-if-->
      </button></li>
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75">
        <!--v-if-->
        <!--v-if--><span class=""><!--v-if--> Settings</span><span class="inline-flex items-center font-medium text-xs px-2 py-1 bg-primary-500 dark:bg-primary-400 text-white dark:text-gray-900 flex-shrink-0 ml-auto relative rounded">badge</span>
      </button></li>
    <li class=""><button type="button" class="group relative w-full flex items-center gap-1.5 px-2.5 py-3.5 rounded-md font-medium text-sm focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 disabled:cursor-not-allowed disabled:opacity-75"><span class="relative inline-flex items-center justify-center flex-shrink-0 rounded-full h-8 w-8 text-sm flex-shrink-0"><img class="rounded-full h-8 w-8 text-sm" alt="nuxtui" src="https://repository-images.githubusercontent.com/428329515/43fec891-9030-4601-8233-5d45ba5c6013"><!--v-if--></span>
        <!--v-if--><span class=""><!--v-if--> About</span>
        <!--v-if-->
      </button></li>
  </ul>
</nav>"
`;
